swap/src/bin/swap_cli.rs

```diff
@@ -320,3 +305,133 @@ async fn init_monero_wallet(
 
     Ok(monero_wallet)
 }
+
+async fn determine_btc_to_swap(
+    request_quote: impl Future<Output = Result<BidQuote>>,
+    initial_balance: impl Future<Output = Result<bitcoin::Amount>>,
+    get_new_address: impl Future<Output = Result<bitcoin::Address>>,
+    wait_for_deposit: impl Future<Output = Result<bitcoin::Amount>>,
+    max_giveable: impl Future<Output = Result<bitcoin::Amount>>,
+) -> Result<bitcoin::Amount> {
+    debug!("Requesting quote");
+
+    let bid_quote = request_quote.await.context("failed to request quote")?;
+
+    info!("Received quote: 1 XMR ~ {}", bid_quote.price);
+
+    // TODO: Also wait for more funds if balance < dust
+    let initial_balance = initial_balance.await?;
+
+    if initial_balance == Amount::ZERO {
+        info!(
+            "Please deposit the BTC you want to swap to {} (max {})",
+            get_new_address.await?,
+            bid_quote.max_quantity
+        );
+
+        let new_balance = wait_for_deposit.await?;
+
+        info!("Received {}", new_balance);
+    } else {
+        info!("Found {} in wallet", initial_balance);
+    }
+
+    let max_giveable = max_giveable.await?;
+    let max_accepted = bid_quote.max_quantity;
+
+    if max_giveable > max_accepted {
+        info!(
+            "Max giveable amount {} exceeds max accepted amount {}!",
```

Comment:
I think it should still say that it will default to the max accepted for the swap!

Reply:
Well, the very next log message will be the spot price one where the user will see how much is being swapped. I also don't want to overload them with information ...

Reply:
Fair enough - let's just wait for feedback and then change things :)